File: GFG POTD/022-September-2025/009-assign-mice-holes.cpp
// https://www.geeksforgeeks.org/problems/assign-mice-holes3053/1




#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    // TC : O(N*log(N))
    // SC : O(1)
    int assignHole(vector<int>& mices, vector<int>& holes) {
        sort(begin(mices), end(mices));
        sort(begin(holes), end(holes));
        
        int minTime = INT_MIN;
        for (int i = 0; i < mices.size(); i ++) {
            minTime = max(minTime, abs(holes[i] - mices[i]));
        }
        return minTime;
    }
};

File: GFG POTD/022-September-2025/009-assign-mice-holes.java
// https://www.geeksforgeeks.org/problems/assign-mice-holes3053/1/




import java.util.*;

class Solution {
    // TC : O(N * log(N))
    // SC : O(1)
    public int assignHole(int[] mices, int[] holes) {
        Arrays.sort(mices);
        Arrays.sort(holes);
        
        int minTime = Integer.MIN_VALUE;
        for (int i = 0; i < mices.length; i ++) {
            minTime = Math.max(minTime, Math.abs(mices[i] - holes[i]));
        }
        return minTime;
    }
};